### PR TITLE
Implement Mask-Guided Background Suppression for GR00T

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
Enhance GR00T

## Summary
- Add optional mask inputs to the VLA data pipeline.
- Introduce mask-driven augmentations (random tint + background noise).
- Enable mask-aware preprocessing in the GR00T N1.6 processor and config.

## Motivation
This PR targets robustness under unseen environments. In action data, only part of the video is directly relevant to the action (like robot arm and apple in the above figure); irrelevant regions can cause spurious correlations and overfitting. By focusing the model on the effective regions via mask-guided augmentations, the same action should remain reliable when the background changes.

If users can identify which video regions are task-relevant versus background and provide masks accordingly, we can apply targeted augmentations during training to encourage the model to ignore irrelevant background. For simulation data, object masks are directly available. For real data, masks can be obtained by training a lightweight segmenter or using off-the-shelf models (e.g., SAM2/SAM3). In our experiments, SAM3 performs well with simple prompt tuning.

## Benchmark

In my simulation experiments, training data were collected in a fixed environment. For evaluation, we constructed fully unseen environments by replacing the wall, shelf, and floor with different appearances.
Performance is measured by rollouts: we run 160 evaluation episodes per setting to benchmark robustness under these seen/partially seen/unseen backgrounds. The following shows preliminary results (no further parameter tuning).

sim

For real-world evaluation, we run 10 rollouts per setting. Training data are collected in a fixed environment; testing is performed both in the same environment and in a visually different one to assess generalization.

real

## What’s Included
- Dataset loaders now optionally read mask modalities aligned with video frames.
- Processor accepts masks and forwards them through the pipeline.
- Mask-based transforms in the Albumentations pipeline:
  - **BackgroundNoiseTransform**: applies random noise to specified mask values.
  - **MaskedColorTransform**: applies random tint to specified mask values.

## Notes
- Masks are optional: behavior remains unchanged if masks are absent.
- Augmentations are controlled via `extra_augmentation_config`.
- Mask augmentations require `use_albumentations_transforms=True`.

## Before submitting

- [x] I've read and followed all steps in the [Making a pull request](https://github.com/NVIDIA/Isaac-GR00T/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [x] I've updated or added any relevant docstrings.
- [x] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

### CODE REVIEW

**Comment by binliunls** [comment]
Hi @yiheng-wang-nv , May I ask if the model trained in this PR can be loaded and rolled out with the main branch code? I remembered there were some gaps of loading model before.

**Comment by yiheng-wang-nv** [comment]
> Hi @yiheng-wang-nv , May I ask if the model trained in this PR can be loaded and rolled out with the main branch code? I remembered there were some gaps of loading model before.

Hi @binliunls , thanks for the question. I refactor the code, and the trained model works fine on the main branch.

**Comment by youliangtan** [comment]
Do you have an example dataset? or a quick code snippet here to test/show example with.

**Comment by yiheng-wang-nv** [comment]
> Do you have an example dataset? or a quick code snippet here to test/show example with.

Hi @youliangtan @XYZ-99  I added a demo dataset (modified from `demo_data`, I keep the first episode, and produce masks) and the corresponding example with tests.
Can you help to re-review the PR? Thanks!

An example output (middle is the processed image):

frame_0283

**Comment by XYZ-99** [comment]
LGTM! @youliangtan WDYT?

**Comment by youliangtan** [comment]
@yiheng-wang-nv I generally think this is good! however one concern is that the npz mask file `episode_000000_masks.npz` is very custom, where there's a custom impl to generate that file after using standard SAM or other similar impl.  Merging also means providing an example and "soft support" this feature. Maybe can write a short desc of how the masks are generated in `examples/mask-guided-background-...`. Let me know what you think.

**Comment by yiheng-wang-nv** [comment]
> @yiheng-wang-nv I generally think this is good! however one concern is that the npz mask file `episode_000000_masks.npz` is very custom, where there's a custom impl to generate that file after using standard SAM or other similar impl. Merging also means providing an example and "soft support" this feature. Maybe can write a short desc of how the masks are generated in `examples/mask-guided-background-...`. Let me know what you think.

Hi @youliangtan , thanks for the suggestion. A new commit is pushed with mask generation related info. Can you help to review it again?

File: examples/mask-guided-background-suppression/README.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+Mask-guided augmentations leverage per-frame segmentation masks to apply targeted image transformations during training. This enables **domain randomization** on specific regions (e.g., replacing backgrounds with noise, tinting foreground objects) without affecting the rest of the image.
+
+This feature is controlled via the `--extra_augmentation_config` argument, which accepts a JSON string specifying which mask regions to augment and how.
+
+---
+
+## Prerequisites
+
+1. **Segmentation masks** must be pre-generated and stored alongside your dataset. The dataset's `info.json` must include a `mask_path` template, and `modality.json` must define a `"mask"` section mapping camera views.
```
Comment (youliangtan) [inline examples/mask-guided-background-suppression/README.md]: @yiheng-wang-nv i didnt see the doc on generating the info here.

Comment (yiheng-wang-nv) [reply examples/mask-guided-background-suppression/README.md]: Oh, @youliangtan , sorry I mis-pushed into another branch (which I used for more tests), now it's added https://github.com/NVIDIA/Isaac-GR00T/pull/521/commits/d361c1b01978bd92b8d37fa517b70397f100a872

**Comment by Nic-Ma** [comment]
Hi @youliangtan , could you please help review it again? Thanks.

**Review by youliangtan — APPROVED** [review]
LGTM. Thanks for adding the additional desc
